### DISCUSSION TITLE
Not getting to repl via stlink on nucleo f401re

### DISCUSSION THREAD

**Discussion by ramprao** [micropython/micropython#15739] [STM32 / Pyboard] [answered]
I just got a nucleo f401re. I was able to flash micropython via the st-flash (built in stlink). But not able to access the board filesystem/repl on the usb port using both rshell and mpremote.

**Comment by davefes** [comment]
On the WeACT BlackPill STM32F411CE the repl is available on UART1 (A9 and A10) even though it has a USB-C connector.  Maybe, the same applies to your board?

**Reply by ramprao** [reply]
Thanks Dave. 
I looked at the mpconfigport.h and UART2 is being used for this board. 

~~I also tried an stm32cubeide program   to access the uart and not able to see it there as well. So puzzled.~~

**Reply by davefes** [reply, marked as answer]
The `mpconfigboard.h` that I am using has:
```
// UART config, 
// 12.5 Mbit/s for USART1 and USART6
// 6.25 Mbit/s for USART2
// Not defining the NAMEs reduces firmware size
// #define MICROPY_HW_UART1_NAME       "USART1"
#define MICROPY_HW_UART1_TX         (pin_A9) // Valid: PA9, PA15, PB6
#define MICROPY_HW_UART1_RX         (pin_A10) // Valid: PA10, PB3, PB7
// #define MICROPY_HW_UART2_NAME       "USART2"
#define MICROPY_HW_UART2_TX         (pin_A2) // Valid: PA2
#define MICROPY_HW_UART2_RX         (pin_A3) // Valid: PA3
// #define MICROPY_HW_UART6_NAME       "USART6"
#define MICROPY_HW_UART6_TX         (pin_A11) // Valid: PA11
#define MICROPY_HW_UART6_RX         (pin_A12) // Valid: PA12

#define MICROPY_HW_UART_REPL        PYB_UART_1
#define MICROPY_HW_UART_REPL_BAUD   115200
```
I think `define MICROPY_HW_UART_REPL        PYB_UART_1` is what is needed ... at least for me.

**Reply by ramprao** [reply]
For my board, its defined as PYB_UART_2. I'll try and rebuild the FW with the REPL UART and connect via USB/UART device and see if there's any luck with it.

**Reply by davefes** [reply]
Good luck.  I am working with this distro:
https://github.com/nspsck/STM32F411CEU6_BlackPill_Micropython
maybe the instructions help.

**Reply by ramprao** [reply]
Thanks for that repo. Will look at it.

**Reply by ramprao** [reply]
Thanks for that tip. For my board in the official repo, there's no UART1. After defining it, i had to change the Pins from what's given above. A9/A10 didn't work for me but B6/A10 did. I also tried UART6 but no luck there as well. 

What's odd is that the UART2 works via the STLINK for STM32CubeIDE.

**Comment by victorallume** [comment]
Which USB port are you plugged in to?

**Reply by ramprao** [reply]
I'm connected to the Mini USB port which is also the STLink port for this board. From what I can see of the config files, the REPL is supposed to be available on this UART.
